src/notification/publish/publish.test.ts

```diff
@@ -1,28 +1,121 @@
-  it(
-    'should return the expected response for /publish/:id',
-    async () => {
-      const response = await request(server).get('/publish/3');
-
-      // Check if the server returns a 200 status code
-      expect(response.status).toBe(200);
-
-      // Check if the response body matches the expected format
-      expect(response.body).toEqual({ id: '3', message: 'From Publish Service' });
-    },
-    // Set the timeout to 20 seconds
-    20000
+// Helper type to mock all possible overloads
+type MockMqttClient = Partial<{
+  on: jest.Mock;
+  publish: jest.Mock;
+  end: jest.Mock;
+}>;
+
+const mockClient = (): MockMqttClient => {
+  const client = {} as MockMqttClient;
+  
+  client.on = jest.fn();
+  
+  client.publish = jest.fn(
+    (topic: string, message: string | Buffer, optsOrCallback?: IClientPublishOptions | PacketCallback, callback?: PacketCallback) => {
+      if (typeof optsOrCallback === 'function') {
+        optsOrCallback(); // Simulate success
+      } else if (callback) {
+        callback(); // Simulate success
+      }
+      return client as unknown as MqttClient; // Ensure it matches the return type
+    }
   );
+
+  client.end = jest.fn(
+    (forceOrCallback?: boolean | (() => void), callback?: (() => void)) => {
+      if (typeof forceOrCallback === 'boolean') {
+        if (callback) callback(); // Simulate success if callback is provided
+      } else if (typeof forceOrCallback === 'function') {
+        forceOrCallback(); // Simulate success
+      } else if (callback) {
+        callback(); // Simulate success if only callback is provided
+      }
+      return client as unknown as MqttClient; // Ensure it matches the return type
+    }
+  );
+
+  return client;
+};
+
+describe('Publish Service', () => {
+  let app: express.Express;
+  const topic = 'test/topic';
+  const options = {
+    username: 'testUser',
+    password: 'testPassword',
+    clientId: 'testClient',
+  };
+
+  beforeEach(() => {
+    app = express();
+
+    (mqtt.connect as jest.Mock).mockReturnValue(mockClient());
+    (connectToMongoDB as jest.Mock).mockResolvedValue(undefined);
+
+    app.get('/publish/:id', async (req, res) => {
+      const client: MqttClient = mqtt.connect('mqtt://test-broker', options);
+      const event = {
+        id: req.params.id,
+        message: 'From Publish Service',
+      };
+
+      client.on('connect', () => {
+        log('Broker connected');
+        client.publish(topic, JSON.stringify(event), {}, (err) => {
+          if (err) {
+            error(`Error publishing message: ${err}`);
+            res.status(500).json({ error: 'Internal Server Error' });
+          } else {
+            client.end();
+            res.json(event);
+          }
+        });
+      });
+
+      client.on('error', (error: Error) => {
+        log(error);
+        res.status(500).json({ error: 'Internal Server Error' });
+      });
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
```

Comment:
@KevinLemon112 I believe you need to close the express server when the tests finish.  Look at this stackoverflow for more info.
https://stackoverflow.com/questions/59011575/how-to-close-express-server-inside-jest-afterall-hook